File: 프로그래머스/LV1/코딩테스트 연습/연습문제/문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    x, x_cnt = 0, 0

    for c in s:
        if x == 0:
            x = c
            x_cnt = 1
            continue   

        x_cnt += 1 if x == c else -1
        
        if x_cnt == 0:
            answer += 1
            x = 0

    if x_cnt != 0:
        answer += 1
    return answer

File: 프로그래머스/LV1/코딩테스트 연습/연습문제/문자열 나누기/문자열 나누기.cpp
#include <string>

using namespace std;

int solution(string s)
{
    int answer = 0;
    int x_cnt = 0;
    char x = 0;


    for (char c : s)
    {
        if (x == 0)
        {
            x = c;
            x_cnt = 1;
            continue;
        }

        x_cnt += x == c ? 1 : -1;

        if (x_cnt == 0)
        {
            answer++;
            x = 0;
        } 
    }
    if (x_cnt != 0)answer++;
    return answer;
}